packages/eas-cli/src/commandUtils/builds.ts

```diff
@@ -0,0 +1,91 @@
+import chalk from 'chalk';
+
+import {
+  AppPlatform,
+  Build,
+  BuildFilter,
+  BuildFragment,
+  BuildStatus,
+  InputMaybe,
+} from '../graphql/generated';
+import { BuildQuery } from '../graphql/queries/BuildQuery';
+import { appPlatformEmojis } from '../platform';
+import { ExpoGraphqlClient } from './context/contextUtils/createGraphqlClient';
+
+export async function ensureBuildExistsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  buildId: string
+): Promise<void> {
+  try {
+    await BuildQuery.byIdAsync(graphqlClient, buildId);
+  } catch {
+    throw new Error(`Couldn't find a build matching the id ${buildId}`);
+  }
+}
+
+export async function fetchBuildsAsync({
+  graphqlClient,
+  projectId,
+  filters,
+}: {
+  graphqlClient: ExpoGraphqlClient;
+  projectId: string;
+  filters?: { statuses?: BuildStatus[]; platform?: string; profile?: string };
+}): Promise<BuildFragment[]> {
+  let builds: BuildFragment[];
+  const queryFilters: InputMaybe<BuildFilter> = {};
+  if (filters?.platform && filters?.platform !== 'all') {
+    queryFilters['platform'] = filters?.platform.toUpperCase() as AppPlatform;
```

Comment:
What do you think about using some mapping from one type to the other type here? By doing this TS will warn us if something type-related changes and it will be more explicit (for example we will see if due to some changes the `toUpperCase` isn't enough)